saleor/graphql/core/mutations.py

```diff
@@ -65,146 +47,159 @@ def _update_mutation_arguments_and_fields(cls, arguments, fields):
-    def __init_subclass_with_meta__(
-            cls, arguments=None, form_class=None, return_field_name=None,
-            _meta=None, exclude=None, **options):
-        if not form_class:
-            raise ImproperlyConfigured(
-                'form_class are required for ModelFormMutation')
-
-        _meta = ModelFormMutationOptions(cls)
-        model = form_class._meta.model
-        if not return_field_name:
-            return_field_name = get_model_name(model)
-        fields = get_output_fields(model, return_field_name)
-        # get mutation arguments based on model form
-        arguments = convert_form_fields(form_class, exclude)
+    def _construct_instance(cls, instance, cleaned_data):
+        """
+        Construct and return a model instance from ``cleaned_data``, but do
+        not save the returned instance to the database.
+        """
+        from django.db import models
+        opts = instance._meta
 
-        _meta.form_class = form_class
-        _meta.model = model
-        _meta.return_field_name = return_field_name
+        for f in opts.fields:
+            if not f.editable or isinstance(f, models.AutoField) or f.name not in cleaned_data:
+                continue
+            if f.name in cleaned_data and cleaned_data[f.name] is None:
+                continue
+            else:
+                f.save_form_data(instance, cleaned_data[f.name])
 
-        super().__init_subclass_with_meta__(_meta=_meta, **options)
+        return instance
 
-        cls._update_mutation_arguments_and_fields(
-            arguments=arguments, fields=fields)
+    @classmethod
+    def _clean_instance(cls, instance, errors):
+        try:
+            instance.full_clean()
+        except ValidationError as validation_errors:
+            message_dict = validation_errors.message_dict
+            for field in message_dict:
+                for message in message_dict[field]:
+                    cls.add_error(errors, field, message)
+        return errors
 
     @classmethod
-    def get_form_kwargs(cls, root, info, **input):
-        return {'data': input}
+    def _save_m2m(cls, instance, cleaned_data):
+        opts = instance._meta
+        for f in chain(opts.many_to_many, opts.private_fields):
+            if not hasattr(f, 'save_form_data'):
+                continue
+            if f.name in cleaned_data and cleaned_data[f.name] is not None:
+                f.save_form_data(instance, cleaned_data[f.name])
 
     @classmethod
-    def save(cls, root, info, **kwargs):
-        """
-        This method is designed to handle form and special cases for easier
-        use of mutate method.
-        """
-        form_kwargs = cls.get_form_kwargs(root, info, **kwargs)
-        form = cls._meta.form_class(**form_kwargs)
-        cls._form = form
-        if form.is_valid():
-            instance = form.save()
-            return instance
-        return None
+    def user_is_allowed(cls, user, input):
+        return True
```

Comment:
I think there should be some kind of check here.

Reply:
I'll add docs to these methods.